c_glib/arrow-glib/file-system.cpp

```diff
@@ -0,0 +1,1050 @@
+  case PROP_FILE_SYSTEM:
+    priv->file_system =
+      *static_cast<std::shared_ptr<arrow::fs::FileSystem> *>(g_value_get_pointer(value));
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_file_system_get_property(GObject *object,
+                                guint prop_id,
+                                GValue *value,
+                                GParamSpec *pspec)
+{
+  switch (prop_id) {
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_file_system_init(GArrowFileSystem *object)
+{
+}
+
+static void
+garrow_file_system_class_init(GArrowFileSystemClass *klass)
+{
+  GParamSpec *spec;
+
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->finalize     = garrow_file_system_finalize;
+  gobject_class->set_property = garrow_file_system_set_property;
+  gobject_class->get_property = garrow_file_system_get_property;
+
+  spec = g_param_spec_pointer("file_system",
+                              "FileSystem",
+                              "The raw std::shared<arrow::fs::FileSystem> *",
+                              static_cast<GParamFlags>(G_PARAM_WRITABLE |
+                                                       G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_FILE_SYSTEM, spec);
+}
+
+/**
+ * garrow_file_system_get_type_name:
+ * @file_system: A #GArrowFileSystem.
+ *
+ * Returns: The name of file system type.
+ *
+ * Since: 1.0.0
+ */
+const gchar *
+garrow_file_system_get_type_name(GArrowFileSystem *file_system)
+{
+  auto arrow_file_system = garrow_file_system_get_raw(file_system);
+  return arrow_file_system->type_name().c_str();
+}
+
+/**
+ * garrow_file_system_get_target_stats:
+ * @file_system: A #GArrowFileSystem.
+ * @path: The path of the target.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Get statistics for the given target.
+ *
+ * Any symlink is automatically dereferenced, recursively.
+ * A non-existing or unreachable file returns an OK status and has
+ * a #GArrowFileType of value %GARROW_FILE_TYPE_NON_EXISTENT.
+ * An error status indicates a truly exceptional condition
+ * (low-level I/O error, etc.).
+ *
+ * Returns: (nullable) (transfer full): A #GArrowFileStats
+ *
+ * Since: 1.0.0
+ */
+GArrowFileStats *
+garrow_file_system_get_target_stats(GArrowFileSystem *file_system,
+                                    const gchar *path,
+                                    GError **error)
+{
+  auto arrow_file_system = garrow_file_system_get_raw(file_system);
+  auto arrow_result = arrow_file_system->GetTargetStats(path);
+  if (garrow::check(error, arrow_result, "[filesystem][get_target_stats]")) {
+    auto arrow_file_stats = std::make_shared<arrow::fs::FileStats>(arrow_result.ValueOrDie());
+    return garrow_file_stats_new_raw(&arrow_file_stats);
+  } else {
+    return NULL;
+  }
+}
+
+namespace {
```

Comment:
How about using `static` instead of anonymous namespace (without indent) only for one internal function?